plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/HistoryTable.java

```diff
@@ -77,14 +76,13 @@ public RecordCursor cursor(ConnectorTransactionHandle transactionHandle, Connect
 
         Set<Long> ancestorIds = ImmutableSet.copyOf(SnapshotUtil.currentAncestorIds(icebergTable));
         TimeZoneKey timeZoneKey = session.getTimeZoneKey();
-        for (HistoryEntry historyEntry : icebergTable.history()) {
```

Comment:
> In the current version of Iceberg library, when not all the
snapshots are initially loaded (for the REST session catalog),
the `snapshotsLog` does not get reinitialized, causing to
expose only the current snapshots in the `history()` method
of the Iceberg table.

is this a bug in the library?

Reply:
> is this a bug in the library?

Yes, please see the discussion https://apache-iceberg.slack.com/archives/C03LG1D563F/p1683871245426989

Reply:
@findinpath unfortunately the discussion is no longer visible in slack (thread is a bit old).

We recently hit an issue here because we made the assumption that `$history` table only contained snapshots that were at some point or another the current snapshot of a table. 

I wonder what is the difference now between `$snapshots` and `$history`? Since I can get pretty much anything from `$history` using only `$snapshots`.

Reply:
@gustavoatt i don't see the replies anymore on the slack thread and lost context :(